grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -717,6 +773,40 @@ static GrpcStatusFunction toGrpcStatusFunction(
         return interceptors;
     }
 
+    private void collectDecorators(Class<?> clazz, @Nullable String path,
+                                   ServerServiceDefinition serverServiceDefinition) {
+        final String serviceName =
+                path != null ? path : '/' + serverServiceDefinition.getServiceDescriptor().getName();
+        // In gRPC, A method name is unique.
+        final Map<String, Method> methods = new HashMap<>();
+        for (Method method : getAllMethods(clazz, withModifier(Modifier.PUBLIC))) {
+            if (method.getDeclaringClass().equals(clazz)) {
```

Comment:
Should we limit the declaring class to `clazz`?
I think some users might want a hierarchical structure for stub implementations.
For example:
```java
abstract class AbstractMyService extends MyServiceImplBase {
    @Decorator(MyDecorator.class)
    public void foo(...) {
        ...
    }

    public void bar(...) {
        ...
    }
}


class BarService extends AbstractMyService {

    @Decorator(BarDecorator.class)
    @Override
    public void bar(...) {
        ...
    }
}
```

Reply:
We should not limit this.
I supported a hierarchical structure for stub implementations. -> [9f4ad54](https://github.com/line/armeria/pull/4041/commits/9f4ad54b34e678d8f02c026a17ba206d48df33dc)